sunpycube/cube/cube_utils.py

```diff
@@ -636,19 +656,53 @@ def _convert_cube_like_index_to_sequence_indices(cube_like_index, cumul_cube_len
         cube_index = cube_like_index
     else:
         sequence_index = np.where(cumul_cube_lengths <= cube_like_index)[0][-1]
-        cube_index = cube_like_index - cumul_cube_lengths[sequence_index]
-        # sequence_index should be plus one as the sequence_index earlier is previous index.
-        sequence_index += 1
+        # if the cube is out of range then return the last index
+        if cube_like_index > cumul_cube_lengths[-1] - 1:
+            cube_index = cumul_cube_lengths[0] - 1
```

Comment:
If the user enters an exposure number which is out of range of the number of exposures in the CubeSequence, I think an error should be raised rather.